mrmustard/lab/states/base.py

```diff
@@ -387,14 +387,17 @@ def auto_shape(
                     else:
                         ansatz = self.ansatz
                     A, b, c = ansatz.triple
-                    shape = autoshape_numba(
-                        math.asnumpy(A),
-                        math.asnumpy(b),
-                        math.asnumpy(c),
-                        max_prob or settings.AUTOSHAPE_PROBABILITY,
-                        max_shape or settings.AUTOSHAPE_MAX,
-                        min_shape or settings.AUTOSHAPE_MIN,
-                    )
+                    try:
+                        shape = autoshape_numba(
+                            math.asnumpy(A),
+                            math.asnumpy(b),
+                            math.asnumpy(c),
+                            max_prob or settings.AUTOSHAPE_PROBABILITY,
+                            max_shape or settings.AUTOSHAPE_MAX,
+                            min_shape or settings.AUTOSHAPE_MIN,
+                        )
+                    except TracerArrayConversionError:  # pragma: no cover
```

Comment:
Can we create a BackendError so that we don't use anything backend-specific in lab?